internal/mempool/mempool.go

```diff
@@ -274,7 +274,7 @@ func (txmp *TxMempool) CheckTx(
 		return nil
 	}
 
-	reqRes, err := txmp.proxyAppConn.CheckTxAsync(ctx, abci.RequestCheckTx{Tx: tx})
+	reqRes, err := txmp.proxyAppConn.CheckTx(ctx, abci.RequestCheckTx{Tx: tx})
```

Comment:
Needs to be `CheckTxSync` I believe. I'm also worried about how this will change the behavior of applications using the Socket Client.

The comment at the top of this method should also be updated.